app/src/main/java/org/simple/clinic/drugs/selectionv2/entry/confirmremovedialog/ConfirmRemovePrescriptionDialog.kt

```diff
@@ -0,0 +1,109 @@
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.fragment.app.FragmentManager
+import com.jakewharton.rxbinding2.view.RxView
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
+import io.reactivex.schedulers.Schedulers.io
+import io.reactivex.subjects.PublishSubject
+import org.simple.clinic.R
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.bp.entry.confirmremovebloodpressure.ConfirmRemovePrescriptionDialogCreated
+import org.simple.clinic.bp.entry.confirmremovebloodpressure.RemovePrescriptionClicked
+import org.simple.clinic.drugs.PrescriptionRepository
+import org.simple.clinic.widgets.ScreenDestroyed
+import org.simple.clinic.widgets.UiEvent
+import java.util.UUID
+import javax.inject.Inject
+
+class ConfirmRemovePrescriptionDialog : AppCompatDialogFragment() {
+
+  @Inject
+  lateinit var prescriptionRepository: PrescriptionRepository
+
+  @Inject
+  lateinit var controller: ConfirmRemovePrescriptionDialogController
+
+  private val onStarts = PublishSubject.create<Any>()
+  private val screenDestroys = PublishSubject.create<ScreenDestroyed>()
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    val dialog = AlertDialog.Builder(requireContext(), R.style.Clinic_V2_DialogStyle_Destructive)
+        .setTitle(R.string.customprescription_delete_title)
+        .setMessage(R.string.customprescription_delete_message)
+        .setPositiveButton(R.string.customprescription_delete_confirm, null)
+        .setNegativeButton(R.string.customprescription_delete_cancel, null)
+        .create()
+
+    onStarts
+        .take(1)
+        .flatMap { setupDialog() }
+        .takeUntil(screenDestroys)
+        .subscribe { uiChange -> uiChange(this) }
+
+    return dialog
+  }
+
+  private fun setupDialog(): Observable<UiChange> {
+    return Observable.merge(dialogCreates(), screenDestroys, removeClicks())
+        .observeOn(io())
+        .compose(controller)
+        .observeOn(mainThread())
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    TheActivity.component.inject(this)
+  }
+
+  override fun onDestroyView() {
+    super.onDestroyView()
+    screenDestroys.onNext(ScreenDestroyed())
+  }
+
+  override fun onStart() {
+    super.onStart()
+    onStarts.onNext(Any())
+  }
+
+  private fun dialogCreates(): Observable<UiEvent> {
+    val prescriptionUuidToDelete = arguments!!.getSerializable(KEY_PRESCRIPTION_UUID) as UUID
+    return Observable.just(ConfirmRemovePrescriptionDialogCreated(prescriptionUuidToDelete))
+  }
+
+  private fun removeClicks(): Observable<UiEvent> {
+    val button = (dialog as AlertDialog).getButton(DialogInterface.BUTTON_POSITIVE)
+    return RxView.clicks(button).map { RemovePrescriptionClicked }
+  }
+
+  companion object {
+    private const val TAG = "CustomPrescriptionEntryScreenFragment"
```

Comment:
Copy paste error? :)

Reply:
Ohh. That doesn't sound right. The screen can potentially have multiple fragments. The tag is always supposed to identify the fragment and not the caller. That's why the API is also called `FragmentManager.findFragmentByTag()`.